tests/scenario/test_charm.py

```diff
@@ -179,3 +184,22 @@ def test_tracing_storage_is_configured_to_s3_if_s3_relation_filled(
     new_config = get_tempo_config(tempo, context)
     expected_config = Tempo(container).generate_config(["otlp_http"], relation_data)
     assert new_config == expected_config
+
+
+def test_ready_check_on_resume(context, tmp_path, caplog):
+    # GIVEN the charm has no tempo-ready service
+    container, tempo = _tempo_mock_with_initial_config(tmp_path, tempo_ready_svc_exists=False)
+
+    state = State(leader=True, containers=[tempo])
+
+    # WHEN we receive a custom-notice event
+    with caplog.at_level("DEBUG"):
+        os.environ["SCENARIO_SKIP_CONSISTENCY_CHECKS"] = "1"
+        # scenario doesn't play nice in this very edge case
+        context.run(_BoundNotice(Notice(Tempo.tempo_ready_notice_key), tempo).event, state)
+        del os.environ["SCENARIO_SKIP_CONSISTENCY_CHECKS"]
```

Comment:
Minor: consider https://docs.pytest.org/en/stable/reference/reference.html#pytest.MonkeyPatch.setenv instead

Reply:
uh, cool, didn't know about that